lib/protocol/SqlString.js

```diff
@@ -72,20 +73,41 @@ SqlString.arrayToList = function arrayToList(array, timeZone) {
 };
 
 SqlString.format = function format(sql, values, stringifyObjects, timeZone) {
-  values = values == null ? [] : [].concat(values);
+  if (values == null) {
+    return sql;
+  }
 
-  var index = 0;
-  return sql.replace(/\?\??/g, function(match) {
-    if (index === values.length) {
-      return match;
-    }
+  if (!(values instanceof Array)) {
```

Comment:
It looks like `.concat()` uses `Array.isArray()` internally, so no the behaviour would not be the same.

> could we get the best of both cases with `value instanceof Array || Array.isArray(value)`?

Yes, it won't be as fast as just using `instanceof`, but it will still be faster :)

```sh
# values = 'string'
values instanceof Array x 71,716,654 ops/sec ±1.51% (86 runs sampled)
values instanceof Array || Array.isArray(values) x 39,596,842 ops/sec ±0.94% (89 runs sampled)
[].concat(values) x 2,426,460 ops/sec ±0.76% (89 runs sampled)
```